pycytominer_transform/convert.py

```diff
@@ -0,0 +1,582 @@
+
+    # update the record to include the destination path
+    record["destination_path"] = destination_path
+
+    return record
+
+
+@flow
+def to_parquet(  # pylint: disable=too-many-arguments
+    source_path: Union[str, pathlib.Path, Any],
+    dest_path: str,
+    source_datatype: Optional[str] = None,
+    targets: Optional[List[str]] = None,
+    concat: bool = True,
+    infer_common_schema: bool = True,
+    **kwargs,
+) -> Dict[str, List[Dict[str, Any]]]:
+    """
+    Export Arrow data to parquet from dataset groups.
+
+    Args:
+      records: Dict[str, List[Dict[str, Any]]]:
+        Grouped records which include metadata and table data related
+        to files which were read.
+      dest_path: str:
+        Destination where parquet files will be written.
+      concat: bool (Default value = True)
+        Whether to concatenate similar records together as one.
+      infer_common_schema: bool (Default value = True)
+        Whether to infer a common schema when concatenating records.
+
+    Returns:
+      Dict[str, List[Dict[str, Any]]]
+        Grouped records which include metadata about destination filepath
+        where parquet file was written.
+    """
+
+    # gather records to be processed
+    records = gather_records(
+        source_path=source_path,
+        source_datatype=source_datatype,
+        targets=targets,
+        **kwargs,
+    )
+
+    if not isinstance(records, Dict):
+        records = records.result()
+
+    results = {}
+    # for each group of records, map writing parquet per file
+    for record_group_name, record_group in records.items():
+
+        # read files
+        record_group = read_file.map(record=record_group)
+
+        # if the record group has more than one record, we will need a unique name
+        unique_name = False
+        if len(record_group) >= 2:
+            unique_name = True
+
+        # map for writing parquet files with list of files via records
+        destinations = write_parquet.map(
+            record=record_group,
+            dest_path=unmapped(dest_path),
+            unique_name=unmapped(unique_name),
+        )
+
+        if concat:
+            # build a new record group
+            results[record_group_name] = concat_record_group.submit(
+                record_group=destinations,
+                dest_path=dest_path,
+                infer_common_schema=infer_common_schema,
+            )
+        else:
+            results[record_group_name] = destinations
+
+    return {
+        key: value.result()
+        if isinstance(value, PrefectFuture)
+        else [inner_result.result() for inner_result in value]
+        if not isinstance(value, Dict)
+        else value
+        for key, value in results.items()
+    }
+
+
+def convert(  # pylint: disable=too-many-arguments
+    source_path: str,
+    dest_path: str,
+    dest_datatype: Literal["parquet"],
+    source_datatype: Optional[str] = None,
+    targets: Optional[List[str]] = None,
```

Comment:
You could save having to include the `default_targets` argument here and simply set `targets = DEFAULT_TARGETS`. Typically it's not recommended to use a list as a default argument since lists are mutable and the single instance of that string will be shared by all invocations of this function, but since you're not modifying it in this case it's fine. Alternatively, you could check if `targets` is `None` in the body of the function and assign it to `DEFAULT_TARGETS` if so.

A safer option would be to change `DEFAULT_TARGETS` to a tuple, which is an immutable type in Python, and change the type of `target` to `Optional[Iterable[str]]`, or explicitly`Optional[Union[List[str], Tuple[str]]]` if you want to exclude people passing strings, which are both iterable and have strings as elements.

Reply:
Thank you this is a great point! I'll update to use a tuple and default instead.